interfaces/cython/cantera/test/test_convert.py

```diff
@@ -565,7 +566,8 @@ class cti2yamlTest(utilities.CanteraTest):
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
-        cti2yaml.convert(pjoin(cls.cantera_data, 'gri30.cti'), 'gri30.yaml')
+        cti2yaml.convert(Path(cls.cantera_data).joinpath('gri30.cti'),
+                         Path(cls.test_work_dir).joinpath('gri30.yaml'))
```

Comment:
If we're going all in on the use of `pathlib`, why don't we just go ahead and convert the `cantera_data`, `test_work_dir`, and `test_data_dir` variables to instances of `pathlib.Path`?

Reply:
Sure. The test failures are because cti2yaml expects a string as the filename, so I'm working on making that pathlib compatible.

Reply:
So at the moment, most of the other test cases assume that `cantera_data` et al. are strings, and pass them to various `os.path` functions that only work with strings in Python 3.5. 3.6 upgrades all the `os.path` functions to also work with pathlib-like objects, but since we're still supporting 3.5 and I don't think it makes sense to update all of the ck2cti tests that will be going away, I'm not going to change this for now.